src/integer/mat_z/set.rs

```diff
@@ -62,6 +78,118 @@ implement_for_owned!(Z, MatZ, SetEntry);
+    ///
+    /// let mut matrix = MatZ::new(4, 3).unwrap();
+    /// matrix.swap_columns(0, 2);
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds)
+    /// if one of the given columns is greater than the matrix or negative.
+    fn swap_columns(
+        &mut self,
+        col0: impl TryInto<i64> + Display,
+        col1: impl TryInto<i64> + Display,
+    ) -> Result<(), MathError> {
+        let col0 = evaluate_index(col0)?;
+        let col1 = evaluate_index(col1)?;
+        if col0 >= self.get_num_columns() || col1 >= self.get_num_columns() {
+            return Err(MathError::OutOfBounds(
+                format!("smaller than {}", self.get_num_columns()),
+                if col0 > col1 {
+                    col0.to_string()
+                } else {
+                    col1.to_string()
+                },
+            ));
+        }
+        unsafe { fmpz_mat_swap_cols(&mut self.matrix, null(), col0, col1) }
+        Ok(())
+    }
+
+    /// Swaps two rows of the specified matrix.
+    ///
+    /// Parameters:
+    /// - `row0`: specifies the first row which is swapped with the second one
+    /// - `row1`: specifies the second row which is swapped with the first one
+    ///
+    /// Returns an empty `Ok` if the action could be performed successfully.
+    /// Otherwise, a [`MathError`] is returned if one of the specified rows is not part of the matrix.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::{integer::MatZ, traits::MatrixSwaps};
+    ///
+    /// let mut matrix = MatZ::new(4, 3).unwrap();
+    /// matrix.swap_rows(0, 2);
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds)
+    /// if one of the given rows is greater than the matrix or negative.
+    fn swap_rows(
+        &mut self,
+        row0: impl TryInto<i64> + Display,
+        row1: impl TryInto<i64> + Display,
+    ) -> Result<(), MathError> {
+        let row0 = evaluate_index(row0)?;
+        let row1 = evaluate_index(row1)?;
+        if row0 >= self.get_num_rows() || row1 >= self.get_num_rows() {
+            return Err(MathError::OutOfBounds(
+                format!("smaller than {}", self.get_num_columns()),
+                if row0 > row1 {
+                    row0.to_string()
+                } else {
+                    row1.to_string()
+                },
+            ));
+        }
+        unsafe { fmpz_mat_swap_rows(&mut self.matrix, null(), row0, row1) }
+        Ok(())
+    }
+}
+
+impl MatZ {
+    /// Reverses all columns of the specified matrix.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::MatZ;
+    ///
+    /// let mut matrix = MatZ::new(4, 3).unwrap();
+    /// matrix.reverse_columns();
+    /// ```
+    pub fn reverse_columns(&mut self) {
+        unsafe { fmpz_mat_invert_cols(&mut self.matrix, null_mut()) }
```

Comment:
why can we provide null_mut() as a second parameter. Please add a comment, such that developers can understand